src/test/java/uk/gov/justice/digital/service/DomainRefreshServiceTest.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.justice.digital.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.spark.SparkConf;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.io.TempDir;
+import uk.gov.justice.digital.client.dynamodb.DynamoDBClient;
+import org.junit.jupiter.api.Test;
+import uk.gov.justice.digital.config.ResourceLoader;
+import uk.gov.justice.digital.domains.DomainExecutor;
+import uk.gov.justice.digital.domains.DomainExecutorTest;
+import uk.gov.justice.digital.domains.model.DomainDefinition;
+import uk.gov.justice.digital.domains.service.DomainService;
+import uk.gov.justice.digital.exceptions.DomainExecutorException;
+import uk.gov.justice.digital.repository.DomainRepository;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.file.Path;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DomainRefreshServiceTest extends DomainService {
+
+    @TempDir
+    Path folder;
+
+    protected DomainDefinition getDomain(final String resource) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        final String json = ResourceLoader.getResource(DomainExecutorTest.class, resource);
+        return mapper.readValue(json, DomainDefinition.class);
+    }
+
+    protected static SparkSession getConfiguredSparkSession(SparkConf sparkConf) {
```

Comment:
As per previous comments, this should extend the appropriate test class.

Reply:
done